src/main/scala/com/microsoft/hyperspace/index/sources/delta/DeltaLakeRelation.scala

```diff
@@ -206,7 +214,7 @@ class DeltaLakeRelation(spark: SparkSession, override val plan: LogicalRelation)
         if (equalOrLessThanLastIndex == versions.size - 1) {
           // The given table version is equal or larger than the latest index's.
           // Use the latest version.
-          index
```

Comment:
@imback82 `getLatestLog()` in getIndexes doesn't guarantee the availability & ACTIVE state. Should we also fix that?

Reply:
You mean https://github.com/microsoft/hyperspace/blob/40c52431b7220ed892f6781efef578af6b826fa3/src/main/scala/com/microsoft/hyperspace/index/IndexCollectionManager.scala#L120-L126
?

For the state, it will be handled `.filter(index => states.isEmpty || states.contains(index.state))`? What's the current behavior (error message), if the index data doesn't exist?

Reply:
No errors, but just apply the invalid index version; which results in empty result.

Reply:
> For the state, it will be handled .filter(index => states.isEmpty || states.contains(index.state))? What's the current behavior (error message), if the index data doesn't exist?

Yep, but it cannot return any previous version of each index if the state of the latest version is not in `states`.

Reply:
good catch. if that's the case, we should definitely fix it.

Reply:
But because of that, we could impl gold standard w/o actual data. 
I created the issue last year #206, but it's a backlog due to low importance.  (as Spark also doesn't check the file existence while reading it, though Spark check that when df creation..)
Maybe it would be good if we could check the data directories. 
@imback82 WDYT?

Reply:
Ah OK. I think just checking directories is not sufficient. We probably need to either check all files, or check nothing based on config?

> Yep, but it cannot return any previous version of each index if the state of the latest version is not in `states`.

Sorry, I didn't fully get this part. It would return `Nil` in this case, which is fine?

Reply:
> Sorry, I didn't fully get this part. It would return Nil in this case, which is fine?

if a user cancel the spark job in the middle of optimize (or refresh), the latest state remains "optimizing". 
However, we could use the previous log version if it's eligible.

Reply:
> Ah OK. I think just checking directories is not sufficient. We probably need to either check all files, or check nothing based on config?

But what about this PR? checking all files for every version should burden.

Reply:
Checking just data directory doesn't seem any better than not checking at all. I am fine with not checking at all in this PR. And we can do a follow up to introduce a config for "nocheck", "directory_check", and "file_check" in one shot.

I thought this PR can be just exposing versions, or am I missing something?